dpypelines/s3_tar_received.py

```diff
@@ -1,22 +1,67 @@
+from dpypelines.pipeline.dataset_ingress_v1 import dataset_ingress_v1
+from dpytools.validation.json import validation
+from dpypelines.pipeline.shared import message
+
 def start(s3_object_name: str):
     """
     Handles the required behaviour when recieving a tar file indicated
     by an s3 object name.
 
-    Example s3_object_name: my-bucket/my-data.tar
-    """
+    Args:
+        s3_object_name (str): The S3 object name of the tar file to be processed.
+        Example: 'my-bucket/my-data.tar'
 
-    # IN ALL CASES, the FOLLOWING STEPS WILL BE IN TRY CATCH
-    # BLOCKS AND WILL NOTIFY RELEVANT PARTIES IN EVENT OF FAILURE
+    """
 
-    # decompress tar file to workspace
+    # Decompress the tar file to the workspace
+    try:
+        decompress_s3_tar(s3_object_name, "input")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to decompress tar file {s3_object_name}", err))
+        raise err
+    
+    # Create a local directory store using the decompressed files
+    try:
+        localStore = LocalDirectoryStore(f"inputs")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to create local directory store at inputs", err))
+        raise err
 
-    # confirm we have a config
+    # Check for the existence of a configuration file
+    try:
+        if not localStore.has_lone_file_matching("pipeline-config.json"):
+            notify.data_engineering(message.unexpected_error("Missing pipeline-config.json")) 
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Error while checking for pipeline-config.json", err))
+        raise err
 
-    # if we dont have a config - make one? (tbd)
+    # Load the configuration file and validate it against a schema
+    try:
+        config_dict = localStore.get_lone_matching_json_as_dict("pipeline-config.json")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Error while getting pipeline-config.json", err))
+        raise err
+    
+    # Retrieve the path to the schema for the configuration
+    try:
+        path_to_schema = get_config_schema_path(config_dict)
+    except Exception as err:
+        notify.data_engineering(message.cant_find_schema(config_dict, err))
+        raise err
 
-    # use config schema to confirm that config is valid
+    # Validate the configuration against the retrieved schema
+    try:    
+        validation.validate_json_schema(schema_path=path_to_schema, data_dict=config_dict, error_msg="Validating pipeline-config.json", indent=2)
+    except Exception as err:
+        notify.data_engineering(message.invalid_config(config_dict, err))
+        raise err
 
-    # use the config info to select and run the appropriate pipeline,
-    # there should literally by a field in the pipeline-config telling
-    # us which pipeline from ./pipeline/* to call.
+    # if we dont have a config - make one? (tbd)
```

Comment:
getting wires crossed, can you just call `https://github.com/ONSdigital/dp-data-pipelines/blob/0682f4e8ca72fc5309bbd032c56a85acceeedb92/dpypelines/pipeline/dataset_ingress_v1.py#L9` here to finish this script please.

to explain, the logic for this initial script is _intended_ (long term vision) to be:

- 1.) check there's a config
- 2.) use the config to call the _appropriate_ secondary script

its to account for the fact that (over time) a tar file and a submitted dataset wont necessarily always be the same thing.

you've done 1 and we're _not_ gonna do anything clever with 2 just yet (as we don't need to, there's only one option for the next step to call) so just explicitly call it for now (without a try catch please)